packages/shared/src/hooks/useConditionalFeature.ts

```diff
@@ -0,0 +1,37 @@
+import { useQuery } from '@tanstack/react-query';
+import { JSONValue, WidenPrimitives } from '@growthbook/growthbook';
+import { useGrowthBook } from '@growthbook/growthbook-react';
+import { generateQueryKey, RequestKey } from '../lib/query';
+import { Feature } from '../lib/featureManagement';
+import { disabledRefetch } from '../lib/func';
+
+export const useConditionalFeature = <T extends JSONValue>({
+  feature,
+  shouldEvaluate,
+}: {
+  feature: Feature<T>;
+  shouldEvaluate: boolean;
+}): WidenPrimitives<T> => {
+  const growthbook = useGrowthBook();
+  const queryKey = generateQueryKey(
+    RequestKey.Feature,
+    null,
+    feature.id,
+    shouldEvaluate,
+  );
```

Comment:
Should it not be per user eg. for logouts? Also why is `shouldEvaluate`  needed, it is not used in queryFn so it just makes additional query key while disabled from what I can see.